addons/l10n_ec/models/res_partner.py

```diff
@@ -63,26 +62,25 @@ def check_vat(self):
                     it_ruc.id,
                     it_dni.id,
                 ):
+                    partner.l10n_ec_vat_error = ''
                     if partner.l10n_latam_identification_type_id.id == it_dni.id and len(partner.vat) != 10:
-                        raise ValidationError(_('If your identification type is %s, it must be 10 digits')
-                                              % it_dni.display_name)
+                        partner.l10n_ec_vat_error = _('If your identification type is %s, it must be 10 digits')\
```

Comment:
l10n_ec_vat_error should be a non-stored computed field with its own compute instead of doing it here.

Reply:
The number of digits should be a hard validation.